cmd/utils/okteto.go

```diff
@@ -23,10 +23,22 @@ import (
 	"github.com/okteto/okteto/pkg/model"
 	"github.com/okteto/okteto/pkg/okteto"
 	"github.com/okteto/okteto/pkg/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-// HasAccessToNamespace checks if the user has access to a namespace/preview
-func HasAccessToNamespace(ctx context.Context, namespace string, oktetoClient types.OktetoInterface) (bool, error) {
+// HasAccessToK8sClusterNamespace checks if the user has access to a namespace
+var HasAccessToK8sClusterNamespace = func(ctx context.Context, namespace string, k8sClient kubernetes.Interface) (bool, error) {
+	_, err := k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
```

Comment:
As we only need the function to get the namespace, narrowing the `k8sClient` dependency into just a `namespaceGetter func(context.Context, string)` would be beneficial for testing and for having better code dependencies

Reply:
we have already discussed this in private but just to put it down:

the ideal would be to make use of interfaces to solve this problem but because both clients do not expose the same functionality this is not the best case with the current design. On the other hand, I think the main thing and a good starting point is to have adequate test coverage on this issue, as agreed: check that we will work with the right client in each case and that the check of whether it is accessible to the namespace is correct, both of which are included in the test. As we have seen, injecting the method to check if a k8s cluster is accessible to the namespace does not allow a simple and clean test and implies perhaps adding some complexity to the logic that the issue does not require. For the moment I would stick with the option that gives us a simple code that covers all cases. Perhaps opening an issue once everything is clearer and the benefits are clearly seen may be a good option!

Reply:
As Adrian said, since the most important part is to check the client creation, we need to mock the `k8sclient` and the `oktetoclient`.

@AdrianPedriza Reading the other test in the same file I found that we can mock the k8sClient like this:
```
	fakeCtxCommand := newFakeContextCommand(fakeOktetoClient, user, []runtime.Object{
		&corev1.Namespace{
			ObjectMeta: v1.ObjectMeta{
				Name: "test",
			},
		},
	})
```

This way we don't need to have the `HasAccessToK8sClusterNamespace` function as a variable

Reply:
That's right. Thank you @AgustinRamiroDiaz ;)